### PR TITLE
Add tests that run example python scripts

### LINKED ISSUE #970: make python examples using_qiskit* and other examples run as a test
### What is the expected enhancement?
Since the using_qiskit are to be use case examples we need to make them part of the tests. We would have found out that job.done was used and I'm hoping user did note make examples based on this when we make this release.

### LINKED ISSUE #2549: example teleport and stochastic_swap are broken.
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?

Examples give error `AttributeError: 'tuple' object has no attribute 'register'`

### What is the expected behavior?

Examples to run

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds a new test module that will execute all the example
scripts to verify they still work as we make changes. However, a number
of example scripts rely on using the ibmq service which is an optional
component now. To make filtering the example scripts between those which
don't need ibmq from those that need ibmq this also moves all the ibmq
examples into a self contained subdir in examples/python. Those examples
which require ibmq are run conditionally as online_tests. For example
scripts which were running on both local simulators and ibmq these are
split into 2 files one for just simulation and the other in ibmq that
uses the ibmq backends.

### Details and comments

Fixes #970
Fixes #2549

### CODE REVIEW

**Comment by ajavadia** [comment]
I would say that we should not introduce a dependency on `ddt` here.

For one, the ddt docs say that the package is only tested for Python 2.7 and 3.3.

But more importantly, I would say that if the standard library provides a functionality, it is strongly preferred over 3rd party libraries. Since Python 3.4, `unittest.subTest` was introduced, which seems to achieve most of what is intended here (i.e. test each example).

**Comment by mtreinish** [comment]
> I would say that we should not introduce a dependency on `ddt` here.
> 
> For one, the ddt docs say that the package is only tested for Python 2.7 and 3.3.
> 
> But more importantly, I would say that if the standard library provides a functionality, it is strongly preferred over 3rd party libraries. Since Python 3.4, `unittest.subTest` was introduced, which seems to achieve most of what is intended here (i.e. test each example).

The docs are out of date they run on 2.7, 3.5, and 3.4 for ci: https://travis-ci.org/datadriventests/ddt/builds/497695081?utm_source=github_status&utm_medium=notification but it's not like the library is particularly complex and it doesn't have a lot of api churn that will likely cause an issues on newer versions of unittest in stdlib: https://github.com/datadriventests/ddt/blob/master/ddt.py

For this use case subtest would work easily enough, where ddt is more useful is in examples like what @1ucian0 is doing in #2563

**Comment by ajavadia** [comment]
So if subTest does the job here, can we use it? Then I'd block #2563 and #2519 until reaching a consensus on https://github.com/Qiskit/qiskit-terra/issues/2589.

File: examples/python/ghz.py [hunk 1]
```diff
@@ -37,28 +36,9 @@
...
-# Set up the API and execute the program.
-###############################################################
-try:
-    IBMQ.load_accounts()
-except:
-    print("""WARNING: No valid IBMQ credentials found on disk. 
-             You must store your credentials using IBMQ.save_account(token, url). 
-             For now, there's only access to local simulator backends...""")
-
 # First version: simulator
```
Comment (lcapelluto) [inline examples/python/ghz.py]: This comment doesn't make sense with line 57 removed

Comment (mtreinish) [reply examples/python/ghz.py]: Yep, good catch. I removed this in: https://github.com/Qiskit/qiskit-terra/pull/2559/commits/38500c5f0601bfe74430b23decb16f8e96373824

**Pull request merged**